server/src/main/java/org/elasticsearch/common/time/EpochMillisDateFormatter.java

```diff
@@ -70,4 +89,31 @@ public String pattern() {
     public DateFormatter parseDefaulting(Map<TemporalField, Long> fields) {
         return this;
     }
+
+    @Override
+    public Locale getLocale() {
+        return locale;
+    }
+
+    @Override
+    public ZoneId getZone() {
+        return zoneId;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(locale);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
```

Comment:
nit: maybe include the usual reference equality shortcut (`if (this == obj) { return true; }`)